src/coreclr/jit/gentree.cpp

```diff
@@ -23482,6 +23509,15 @@ GenTree* Compiler::gtNewSimdShuffleNode(
             // swap the operands to match the encoding requirements
             retNode = gtNewSimdHWIntrinsicNode(type, op2, op1, NI_AVX512BW_PermuteVar32x16, simdBaseJitType, simdSize);
         }
+        else if (elementSize == 1)
+        {
+            assert(compIsaSupportedDebugOnly(InstructionSet_AVX512VBMI));
+            op2                        = gtNewVconNode(type);
+            op2->AsVecCon()->gtSimdVal = vecCns;
+
+            // swap the operands to match the encoding requirements
+            retNode = gtNewSimdHWIntrinsicNode(type, op2, op1, NI_AVX512VBMI_PermuteVar64x8, simdBaseJitType, simdSize);
+        }
```

Comment:
Looks like we're missing `else if (elementSize == 2)` ?

Reply:
https://github.com/DeepakRajendrakumaran/runtime/blob/MRT143/src/coreclr/jit/gentree.cpp#L23499-L23517 ?

Reply:
Ah, nevermind. This was already handled just above and I didn't expand the right section to see it.